File: 75-sort-colors/75-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        
        int key;
        int j;
        int n = nums.size();
        for(int i=1;i<n;i++)
        {
            key = nums[i];
            j = i - 1;
            while(j>=0 && nums[j] > key)
            {
                nums[j+1] = nums[j];
                j = j - 1;
            }
            nums[j+1] = key;
            
            
        }
        
        
    }
};

File: 75-sort-colors/75-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int low = 0;
        int n = nums.length;
        int high = n -1;
        int mid = 0;
        int temp = 0;
        while(mid <= high)
        {
            switch(nums[mid])
            {
                case 0:
                    {   
                    temp = nums[mid];
                    nums[mid] = nums[low];
                    nums[low] = temp;
                    mid++;
                    low++;
                    break;
                    }     
                case 1:
                    {
                        mid++;
                        break;
                    }
                case 2:
                    {
                        temp = nums[mid];
                        nums[mid] = nums[high];
                        nums[high] = temp;
                        high--;
                        break;
                    }
                    
            }
        }
        
       
        
    }
}